addons/mail/wizard/mail_compose_message.py

```diff
@@ -328,10 +330,86 @@ def _action_send_mail(self, auto_commit=False):
 
                 result_mails_su += batch_mails_sudo
                 if wizard.composition_mode == 'mass_mail':
+                    wizard._action_send_mail_create_notifications(batch_mails_sudo)
                     batch_mails_sudo.send(auto_commit=auto_commit)
 
         return result_mails_su, result_messages
 
+    def _action_send_mail_create_notifications(self, batch_mails_sudo):
+        """Create notifications for the mails of the batch."""
+        # create mapping from email to partner
+        def parse_email_strings(emails_strings):
+            """Transform multiple email lists into a list of normalized emails.
+
+            :emails_strings [str]: List of strings representing comma-separated emails
+            :return Set[str]: Set of valid normalized emails
```

Comment:
I think in Odoo we always use the type in lower case (so dict, set, etc instead of Dict, Set...)